examples/dune

```diff
@@ -0,0 +1,16 @@
+(executable
+ (name mustache_example)
+ (modules mustache_example)
+ (libraries mustache ezjsonm))
+
+(executable
+ (name mustache_cli)
+ (public_name mustache-cli)
```

Comment:
If this exe is public, I think `examples` is not a good source directory for it anymore.

Reply:
@rgrinberg Is this what needs to be done here? https://github.com/anton-trunov/ocaml-mustache/commit/82957c456d87c81bf2a108ce4d66b150943c7fd9

Reply:
@rgrinberg Thanks for the invitation! Let me push those changes here.